common/core/gpudevice.cpp

```diff
@@ -324,6 +334,38 @@ void GpuDevice::HandleHWCSettings() {
           uint32_t rotation_num = atoi(rotation_str.c_str());
           display_rotation.emplace_back(rotation_num);
           rotation_display_index.emplace_back(physical_index);
+        } else if (!key.compare(key_float_display)) {
+          std::string index_str;
+          std::string float_rect_str;
+          std::vector<int32_t> float_rect;
+          std::istringstream i_value(value);
+
+          // Got display index
+          std::getline(i_value, index_str, ':');
+          if (index_str.empty() || index_str.find_first_not_of("0123456789") !=
+            std::string::npos)
+            continue;
+
+          int32_t index = atoi(index_str.c_str());
+
+          // Got rectangle configuration
+          while (std::getline(i_value, float_rect_str, '+')) {
+            if (float_rect_str.empty() ||
+                float_rect_str.find_first_not_of("0123456789") !=
+                    std::string::npos)
+              continue;
+            size_t float_rect_val = atoi(float_rect_str.c_str());
+
+            // Save the rectangle - left, top, right & bottom
+            float_rect.emplace_back(float_rect_val);
+          }
+
+          // If entire rect is available, store the parameters
+	  // TODO remove hard code
+	  if (float_rect.size() == 4) {
+	    float_display_indices.emplace_back(index);
```

Comment:
indentation issues git diff | clang-format-diff-3.5 -p 1 -style=file -i should fix these